### PR TITLE
[Trainer] Support release gradients for develop branch

### PR DESCRIPTION
**Pull request by haohongxiang**
### PR types

Bug fixes
### PR changes

Others
### Description

[Trainer] Support release gradients for develop branch

This is a cherry-pick PR to develop branch, and the original PR is [PR-7322](https://github.com/PaddlePaddle/PaddleNLP/pull/7322)

### CODE REVIEW

File: paddlenlp/trainer/training_args.py [hunk 1]
```diff
@@ -930,6 +930,7 @@ def __post_init__(self):
                                 "enable_dp_comm_overlap",
                                 "enable_sharding_comm_overlap",
                                 "enable_timer",
+                                "enable_release_grads",
```
Comment (ZHUI) [inline paddlenlp/trainer/training_args.py:934]: 注释里面加一下 这个功能的介绍？

Comment (haohongxiang) [reply paddlenlp/trainer/training_args.py:934]: Done

Comment (ZHUI) [reply paddlenlp/trainer/training_args.py:934]: https://github.com/PaddlePaddle/PaddleNLP/blob/053845a7678efee64c7c0d960790fdffe24fcd4c/paddlenlp/trainer/training_args.py#L580

https://github.com/PaddlePaddle/PaddleNLP/blob/053845a7678efee64c7c0d960790fdffe24fcd4c/paddlenlp/trainer/training_args.py#L238

Comment (ZHUI) [reply paddlenlp/trainer/training_args.py:934]: 是这两个地方

**Review by ZHUI — APPROVED** [review]
LGTM

**Pull request merged**